packages/compass-aggregations/src/components/focus-mode/focus-mode-modal-header.tsx

```diff
@@ -318,6 +328,7 @@ export default connect(
     return {
       stageIndex,
       isEnabled: !stage?.disabled,
+      insight: stage ? getInsightForStage(stage, env) : undefined,
```

Comment:
That's good enough now that you moved insights definition to a const outside of the function, but I'd suggest to pay attention when passing objects like that because this can completely negate any benefits of redux connect if object reference is changing on every re-render

Reply:
Just noticed that it's literally what happens at the next line, so doesn't matter in that case, it's already not memo-ing this component, re-rendering any time any state anywhere in the store changes

Reply:
Ohh wow I definitely missed that. Thank you for pointing it out, I will surely keep this on my watch for further changes.

Reply:
> 

That was a change made by me at some point 😵‍💫 I will fix that right away in a follow up PR separate from this.